eng/mgmt/automation/generate_data.py

```diff
@@ -47,40 +47,17 @@ def sdk_automation_typespec(config: dict) -> List[dict]:
         service = None
         module = None
         try:
-            def remove_prefix(text, prefix):
-                if text.startswith(prefix):
-                    return text[len(prefix):]
-                return text
-
-            def find_sdk_folder():
-                cmd = ['git', 'add', '.']
-                check_call(cmd, sdk_root)
-
-                cmd = ['git', 'status', '--porcelain', '**/tsp-location.yaml']
-                logging.info('Command line: ' + ' '.join(cmd))
-                output = subprocess.check_output(cmd, cwd=sdk_root)
-                output_str = str(output, 'utf-8')
-                git_items = output_str.splitlines()
-                sdk_folder = None
-                if len(git_items) > 0:
-                    tsp_location_item: str = git_items[0]
-                    sdk_folder = tsp_location_item[1:].strip()[0:-len('/tsp-location.yaml')]
-
-                cmd = ['git', 'reset', '.']
-                check_call(cmd, sdk_root)
-
-                return sdk_folder
-
-            repo = remove_prefix(repo_url, 'https://github.com/')
-            cmd = ['npx', 'tsp-client', 'init', '--debug', '--tsp-config', tsp_dir, '--commit', head_sha, '--repo', repo]
-            check_call(cmd, sdk_root)
-
-            sdk_folder = find_sdk_folder()
+            cmd = ['pwsh', './eng/common/scripts/TypeSpec-Project-Process.ps1', tsp_dir, head_sha, repo_url]
```

Comment:
@weidongxu-microsoft why did this need to be reverted? Do we have a plan to switch back?

Reply:
It was due to the failure when the automation runs from a private repo.

PR https://github.com/Azure/azure-rest-api-specs-pr/pull/16496/checks?check_run_id=21889059048

failure
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3520093&view=logs&j=a8a7a537-82b0-583c-7971-bac70b9822ca&t=37e3947b-3cfb-5d36-86ba-0e22bb7dbc33&s=eb9754a7-3885-5b5b-bd91-16c95dd7881e

automation passes after revert PR
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3520093&view=logs&j=a8a7a537-82b0-583c-7971-bac70b9822ca&t=37e3947b-3cfb-5d36-86ba-0e22bb7dbc33&s=eb9754a7-3885-5b5b-bd91-16c95dd7881e

I didn't have time to investigate why it fails, but the request is pretty urgent. Therefore, I did the revert PR first to unblock service.

Reply:
Thanks @weidongxu-microsoft for the context.

@catalinaperalta based on https://github.com/Azure/azure-sdk-tools/issues/7292 I thought we supported the private repo. Is there a bug or more work to be done to enable it?

Reply:
issue on azure-sdk-tools https://github.com/Azure/azure-sdk-tools/issues/8012

Reply:
Will take a look, I missed the original notification on this thread. Thanks for opening an issue!

Reply:
IIRC in the pipeline we'd need to have git credentials configured for the git sparse checkout command to work. A few months ago I spoke to Ben about this since I'd seen this issue https://github.com/Azure/azure-sdk-tools/issues/3222 opened about this exact scenario. I think we'd need to have a PAT or git credentials configured in the pipeline for this to work. Locally for private spec repo support we rely on folks already being logged in with git/having their credentials configured.